src/algorithms/peter_norving.py

```diff
@@ -0,0 +1,165 @@
+class PeterNorvig:
+
+    def __init__(self):
+        self.digits = '123456789'
+        self.rows = 'ABCDEFGHI'
+        self.colums = self.digits
+        self.squares = self.cross(self.rows, self.colums)
+        self.unitlist = [self.cross(self.rows, colum) for colum in
+                         self.colums] + [self.cross(row, self.colums)
+                for row in self.rows] + [self.cross(row_square,
+                colum_square) for row_square in ('ABC', 'DEF', 'GHI')
+                for colum_square in ('123', '456', '789')]
+        self.units = dict((square, [unit for unit in self.unitlist
+                          if square in unit]) for square in
+                          self.squares)
+        self.peers = dict((square, set(sum(self.units[square], []))
+                          - set([square])) for square in self.squares)
+
+    def cross(self, row_key_list, column_key_list):
+        """
+        Cross product of elements in row_key_list and elements in column_key_list.
+
+        Keyword arguments:
+        row_key_list -- Mostly this will be the row values. i.e  1, 2, 3,etc
+        column_key_list -- Mostly this will be the column values. i.e A, B ,C, etc
+        returns list with the filled keys crossed that will be used as
+        coordinates, i.e.  [1A, 2A ].
+        """
+
+        return [key + value for key in row_key_list for value in column_key_list]
+
+    def parse_grid(self, grid):
```

Comment:
Since this is a class modified to be used on the application I will allow just on this class to have different types of return values, usually you should have the same type of return value for your methods, if your method return a Boolean it should return a Boolean for all the cases if returns a string should return an string in all the cases and not for some cases a boolean